src/core/opamSystem.ml

```diff
@@ -175,6 +175,44 @@ let write file contents =
   output_string oc contents;
   close_out oc
 
+let setup_copy ?(chmod = fun x -> x) ~src ~dst () =
+  let ic = open_in_bin src in
+  let oc =
+    try
+      let perm =
+        (Unix.fstat (Unix.descr_of_in_channel ic)).st_perm |> chmod
+      in
+      open_out_gen
+        [ Open_wronly; Open_creat; Open_trunc; Open_binary ]
+        perm dst
+    with exn ->
+      close_in ic;
+      raise exn
+  in
+  (ic, oc)
+
+let copy_channels =
+  let buf_len = 65536 in
```

Comment:
Isn't this a bit large ? Does it get much better perf than, say, a block size of 4096 ?

Reply:
No idea. I took the value from the dune implementation. I don't care enough to benchmark it, so I changed it to the more reasonable-looking 4096.